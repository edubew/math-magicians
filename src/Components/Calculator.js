import React, { Component } from 'react';
import calculate from '../logic/calculate';

export class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
  }

  // Event Listener
  eventHandler = (e) => {
    const value = e.target.textContent;
    this.setState((obj) => calculate(obj, value));
  };

  render() {
    const { next, total, operation } = this.state;
    const result = `${
      total || operation || next
        ? `${total || ''} ${operation || ''} ${next || ''}`
        : 0
    }`;
    return (
      <div className="calculator__container">
        <div className="calculator__output">
          <span>{result}</span>
        </div>
        <button type="submit" onClick={this.eventHandler}>
          AC
        </button>
        <button type="submit" onClick={this.eventHandler}>
          +/-
        </button>
        <button type="submit" onClick={this.eventHandler}>
          %
        </button>
        <button
          type="submit"
          onClick={this.eventHandler}
          className="calculator__operand"
        >
          รท
        </button>
        <button type="submit" onClick={this.eventHandler}>
          7
        </button>
        <button type="submit" onClick={this.eventHandler}>
          8
        </button>
        <button type="submit" onClick={this.eventHandler}>
          9
        </button>
        <button
          type="submit"
          onClick={this.eventHandler}
          className="calculator__operand"
        >
          x
        </button>
        <button type="submit" onClick={this.eventHandler}>
          4
        </button>
        <button type="submit" onClick={this.eventHandler}>
          5
        </button>
        <button type="submit" onClick={this.eventHandler}>
          6
        </button>
        <button
          type="submit"
          onClick={this.eventHandler}
          className="calculator__operand"
        >
          -
        </button>
        <button type="submit" onClick={this.eventHandler}>
          1
        </button>
        <button type="submit" onClick={this.eventHandler}>
          2
        </button>
        <button type="submit" onClick={this.eventHandler}>
          3
        </button>
        <button
          type="submit"
          onClick={this.eventHandler}
          className="calculator__operand"
        >
          +
        </button>
        <button type="submit" onClick={this.eventHandler} className="span__two">
          0
        </button>
        <button type="submit" onClick={this.eventHandler}>
          .
        </button>
        <button
          type="submit"
          onClick={this.eventHandler}
          className="calculator__operand"
        >
          =
        </button>
      </div>
    );
  }
}

export default Calculator;
