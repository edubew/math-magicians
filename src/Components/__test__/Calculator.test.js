/* eslint-disable no-unused-vars */
import {
  cleanup, fireEvent, screen, render,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('Calculator', () => {
  afterEach(cleanup);

  it('renders calculator correctly', () => {
    const tree = renderer.create(<Calculator />);
    expect(tree).toMatchSnapshot();
  });

  it('should fire click event for button 3', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('3'));
    const answer = document.getElementsByClassName('answer');
  });

  it('should display 0 when AC is clicked', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('AC'));
    const answer = document.getElementsByClassName('answer');
  });

  it('should display 1.1 ', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('.'));
    fireEvent.click(screen.getByText('1'));
    const answer = document.getElementsByClassName('answer');
  });
});
