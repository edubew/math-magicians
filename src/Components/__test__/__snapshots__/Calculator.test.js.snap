// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders calculator correctly 1`] = `
<div
  className="calculator__wrapper"
>
  <div
    className="calculator__container"
  >
    <div
      className="calculator__output"
    >
      <span>
        0
      </span>
    </div>
    <button
      onClick={[Function]}
      type="submit"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      %
    </button>
    <button
      className="calculator__operand"
      onClick={[Function]}
      type="submit"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      9
    </button>
    <button
      className="calculator__operand"
      onClick={[Function]}
      type="submit"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      6
    </button>
    <button
      className="calculator__operand"
      onClick={[Function]}
      type="submit"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      3
    </button>
    <button
      className="calculator__operand"
      onClick={[Function]}
      type="submit"
    >
      +
    </button>
    <button
      className="span__two"
      onClick={[Function]}
      type="submit"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      .
    </button>
    <button
      className="calculator__operand"
      onClick={[Function]}
      type="submit"
    >
      =
    </button>
  </div>
  <div>
    <img
      alt="math icon"
      className="calculator__image"
      src="math.jpeg"
    />
  </div>
</div>
`;
